Backend/Remora.Discord.Gateway/Transport/WebSocketPayloadTransportService.cs

```diff
@@ -55,27 +68,41 @@ public class WebSocketPayloadTransportService : IPayloadTransportService, IAsync
     /// <summary>
     /// Initializes a new instance of the <see cref="WebSocketPayloadTransportService"/> class.
     /// </summary>
-    /// <param name="services">The services available to the application.</param>
+    /// <param name="memoryStreamManager">A <see cref="RecyclableMemoryStream"/> pool.</param>
     /// <param name="jsonOptions">The JSON options.</param>
     public WebSocketPayloadTransportService
     (
-        IServiceProvider services,
+        RecyclableMemoryStreamManager memoryStreamManager,
         JsonSerializerOptions jsonOptions
     )
     {
-        _services = services;
+        _memoryStreamManager = memoryStreamManager;
         _jsonOptions = jsonOptions;
+
+        _sendSemaphore = new SemaphoreSlim(1, 1);
```

Comment:
nitpick: Rider should be giving a hint that `new SemaphoreSlim(1)` is equivalent